### ISSUE TITLE
Tests with DataProvider on parallel mode is failing

### ISSUE THREAD

**Issue by SrinivasanTarget** [saikrishna321/AppiumTestDistribution#317]

**Comment by SrinivasanTarget** [comment]
@saikrishna321 looks like things are screwed up :)

**Comment by saikrishna321** [comment]
> if i have both dp and normal, test is failing in distribute mode 

you mean normal test in distribute fails ?

**Comment by SrinivasanTarget** [comment]
not normal in distribute mode... it fails only if you have above combinations

**Comment by SrinivasanTarget** [comment]
might be a testNG issue too

**Comment by shridharkalagi** [comment]
this happens only when `@DataProvider(parallel = true)` and `data-provider-thread-count` is greater than one in `testng.xml` I have created an Issue [here](https://github.com/cbeust/testng/issues/1775) with more details.

**Comment by saikrishna321** [comment]
@SrinivasanTarget  @shridharkalagi from testNg issue comments, its a bug at our end we had to handle the device pool correctly

**Comment by shridharkalagi** [comment]
yes. For now, easy workaround would be to remove `parallel=true` for the dataprovider

**Comment by saikrishna321** [comment]
@shridharkalagi but does not solve the problem 🙅‍♂️

**Comment by saikrishna321** [comment]
@SrinivasanTarget has this been tested ? if yes, can the branch get merged ?

**Comment by SrinivasanTarget** [comment]
Not yet..will keep you posted
